Snowflake.Data/Core/HttpUtil.cs

```diff
@@ -387,8 +387,8 @@ internal RetryHandler(HttpMessageHandler innerHandler, bool disableRetry, bool f
                     await Task.Delay(TimeSpan.FromSeconds(backOffInSec), cancellationToken).ConfigureAwait(false);
                     totalRetryTime += backOffInSec;
                     // Set next backoff time
-                    backOffInSec = backOffInSec >= maxDefaultBackoff ?
-                            maxDefaultBackoff : backOffInSec * 2;
+                    backOffInSec = backOffInSec >= MAX_BACKOFF ?
+                            MAX_BACKOFF : backOffInSec * 2;

```

Comment:
Do we terminate based on max retry? Am I missing any thing here as outer loop is while(true), but I am not seeing terminate based on time or count.

Reply:
Currently the retry on request sending is only limited by timeout.
https://github.com/snowflakedb/snowflake-connector-net/blob/master/Snowflake.Data/Core/HttpUtil.cs#L335
The timeout is set here:
https://github.com/snowflakedb/snowflake-connector-net/blob/master/Snowflake.Data/Core/RestRequester.cs#L173
This is existing behavior has been there for years and won't be a problem in most cases since
- it only retry on retry-able errors https://github.com/snowflakedb/snowflake-connector-net/blob/master/Snowflake.Data/Core/HttpUtil.cs#L367
- we have limited timeout by default
It's good to have a max number of retry though, which I think we could add later.

Reply:
But if we get retry-able errors continuously in the loop and never succeed, that will go in infinite loop. right?

Reply:
Yes, it would be infinite loop if we keep getting retry-able errors. I agree a max number of retry should be added and maybe configurable. But that's existing behavior, not introduced by this change, so I think we should make a different PR for that later.

Reply:
Discuss with Harry to do it as part of another PR.